### PR TITLE
Convert "English-to-Spanish translation with KerasNLP" to keras-core

### PR DESCRIPTION
**Pull request by freedomtan**
1. colab [notebook](https://colab.research.google.com/drive/1VOr8wDFMQQbPoy6tFfAWguZ6GLDtJam1?usp=sharing)
2. diff b8f73d5
  - keras -> keras_croe
  - some tf ops -> keras_core.ops
  - tf.data -> tf_data
  - remove `mask_zero = True` because it causes an error and is not necessary for this example

### CODE REVIEW

File: examples/keras_io/nlp/neural_machine_translation_with_keras_nlp.py [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+it provides the next words in the target sentence -- what the model will try to predict.
+
+We will add special tokens, `"[START]"` and `"[END]"`, to the input Spanish
+sentence after tokenizing the text. We will also pad the input to a fixed length.
+This can be easily done using `keras_nlp.layers.StartEndPacker`.
+"""
+
+
+def preprocess_batch(eng, spa):
+    batch_size = spa.shape[0]
```
Comment (fchollet) [inline examples/keras_io/nlp/neural_machine_translation_with_keras_nlp.py:251]: I'm surprised this works -- I'd assume this needs to be `ops.shape(spa)[0]`

File: examples/keras_io/nlp/neural_machine_translation_with_keras_nlp.py [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+tokens generated so far, until we hit the token `"[END]"`.
+
+For decoding, we will use the `keras_nlp.samplers` module from
+KerasNLP. Greedy Decoding is a text decoding method which outputs the most
+likely next token at each time step, i.e., the token with the highest probability.
+"""
+
+
+def decode_sequences(input_sentences):
+    batch_size = 1
```
Comment (fchollet) [inline examples/keras_io/nlp/neural_machine_translation_with_keras_nlp.py:422]: Why the change compared to the more general `ops.shape(input_sentences)[0]`?

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: examples/keras_io/nlp/neural_machine_translation_with_keras_nlp.py (hunk 1 above)
Comment (freedomtan) [reply examples/keras_io/nlp/neural_machine_translation_with_keras_nlp.py:251]: I can change it to `ops.shape(spa)[0]`. This works because spa is a tf tensor (to process data from tf.data).

File: examples/keras_io/nlp/neural_machine_translation_with_keras_nlp.py (hunk 2 above)
Comment (freedomtan) [reply examples/keras_io/nlp/neural_machine_translation_with_keras_nlp.py:422]: Because I change the what were passed to `decode_sequences()`, `input_sentences` is not a tensor.
Before, the `decode_sequences()`, takes a TF tensor (`decode_sequences(tf.constant([input_sentence]))`), but it seems Torch doesn't have string tensor, I change `decode_sequences()`  to take a list of string.

File: examples/keras_io/nlp/neural_machine_translation_with_keras_nlp.py (hunk 1 above)
Comment (fchollet) [reply examples/keras_io/nlp/neural_machine_translation_with_keras_nlp.py:251]: Let's use `ops.shape(spa)[0]` for better generality.

File: examples/keras_io/nlp/neural_machine_translation_with_keras_nlp.py (hunk 2 above)
Comment (fchollet) [reply examples/keras_io/nlp/neural_machine_translation_with_keras_nlp.py:422]: Got it.

File: examples/keras_io/nlp/neural_machine_translation_with_keras_nlp.py (hunk 1 above)
Comment (freedomtan) [reply examples/keras_io/nlp/neural_machine_translation_with_keras_nlp.py]: done

**Review by fchollet — APPROVED** [review]
LGTM
